packages/unmock-core/src/service/service.ts

```diff
@@ -44,13 +50,15 @@ export class Service implements IService {
     this.name = opts.name;
     // Update the paths in the first level to regex if needed
     if (this.schema === undefined || this.schema.paths === undefined) {
+      this.matcher = new OASMatcher({ schema: this.schema });
```

Comment:
IIRC this doesn't copy `this.schema` into `OASMatcher`, but instead has the same object, right?
If that's the case, one could initialize `this.matcher` after `this.name` I think.

Reply:
Yes that's true, personally I don't like the idea of sharing a mutable object and relying on it becoming updated at a later state but that would work in this case...

Reply:
We can consider it a "one-time-mutable" though.
Alternatively, I can change it so that `Service` holds regex patterns elsewhere and just links to the original path, so the schema is indeed mutable.

Then one would use the `findEndpoint` method in `IService`.

Reply:
Hmm yeah, to me it would be clearest if a (static) call in the constructor would just copy `opts.schema` and add its own fields to the paths if they exist and then set the augmented version to `this.schema` and initialized the matcher with that. But it doesn't matter that much now, I'll just leave this for now if that's ok.

Reply:
Of course.

If the aim is to keep the schema as is, I would rather avoid unnecessary copies and instead just create a `regexPathToPath` dictionary.